kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSPropertyDeclarationImpl.kt

```diff
@@ -107,13 +113,29 @@ class KSPropertyDeclarationImpl private constructor(internal val ktPropertySymbo
 
     override val hasBackingField: Boolean by lazy {
         if (origin == Origin.KOTLIN_LIB || origin == Origin.JAVA_LIB) {
-            val fileManager = ResolverAAImpl.instance.javaFileManager
-            val parentClass = this.findParentOfType<KSClassDeclaration>()
-            val classId = (parentClass as KSClassDeclarationImpl).ktClassOrObjectSymbol.classIdIfNonLocal!!
-            val virtualFileContent = analyze {
-                (fileManager.findClass(classId, analysisScope) as JavaClassImpl).virtualFile!!.contentsToByteArray()
+            when {
+                ktPropertySymbol.receiverParameter != null -> false
+                ktPropertySymbol.initializer is KtConstantInitializerValue -> true
+                (ktPropertySymbol as? KtKotlinPropertySymbol)?.isLateInit == true -> true
+                ktPropertySymbol.modality == Modality.ABSTRACT -> false
+                else -> {
+                    val classId = when (
+                        val containerSource =
+                            (ktPropertySymbol as? KtFirKotlinPropertySymbol)?.firSymbol?.containerSource
+                    ) {
+                        is JvmPackagePartSource -> containerSource.classId
+                        is KotlinJvmBinarySourceElement -> containerSource.binaryClass.classId
+                        else -> null
+                    } ?: return@lazy ktPropertySymbol.hasBackingField
```

Comment:
Should this be fixed in the upstream?

Reply:
unfortunately, the implementation in AA reads from metadata and this information is missing in metadata, an ultimate fix will require fix to metadata in klib which will take a long time.